parser/prism/Translator.cc

```diff
@@ -763,10 +804,14 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
             auto source = parser.extractString(unescaped);
 
             // TODO: can these have different encodings?
-            return make_unique<parser::Symbol>(parser.translateLocation(loc), gs.enterNameUTF8(source));
+            return make_node<parser::Symbol>(parser.translateLocation(loc), gs.enterNameUTF8(source));
         }
         case PM_TRUE_NODE: { // The `true` keyword
-            return translateSimpleKeyword<pm_true_node, parser::True>(node);
+            auto sorbetNode = translateSimpleKeyword<pm_true_node, parser::True>(node);
+
+            sorbetNode->cacheDesugaredExpr(MK::True(sorbetNode->loc));
```

Comment:
Here's a simple example of how migrating a Prism Node to an `ast:ExpressionPtr` would look like. This is a simple case because it's just a leaf node, with not much other information.

We stash this expr into the `NodeAndExpr sorbetNode`, which will later be consumed by either a parent Node (which uses it to construct its own AST expression), or `node2TreeImpl` in `Desugar.cc`.